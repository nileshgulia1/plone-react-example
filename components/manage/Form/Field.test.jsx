import React from 'react';
import renderer from 'react-test-renderer';
import Field from './Field';

jest.mock('../Widgets/ArrayWidget', () =>
  jest.fn(() => <div className="ArrayWidget" />),
);
jest.mock('../Widgets/CheckboxWidget', () =>
  jest.fn(() => <div className="CheckboxWidget" />),
);
jest.mock('../Widgets/PasswordWidget', () =>
  jest.fn(() => <div className="PasswordWidget" />),
);
jest.mock('../Widgets/SelectWidget', () =>
  jest.fn(() => <div className="SelectWidget" />),
);
jest.mock('../Widgets/TextWidget', () =>
  jest.fn(() => <div className="TextWidget" />),
);
jest.mock('../Widgets/TextareaWidget', () =>
  jest.fn(() => <div className="TextareaWidget" />),
);
jest.mock('../Widgets/DatetimeWidget', () =>
  jest.fn(() => <div className="DatetimeWidget" />),
);
jest.mock('../Widgets/WysiwygWidget', () =>
  jest.fn(() => <div className="WysiwygWidget" />),
);

describe('Field', () => {
  it('renders a richtext field', () => {
    const component = renderer.create(<Field widget="richtext" id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a textarea field', () => {
    const component = renderer.create(<Field widget="textarea" id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a datetime field', () => {
    const component = renderer.create(<Field widget="datetime" id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a password field', () => {
    const component = renderer.create(<Field widget="password" id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a text field', () => {
    const component = renderer.create(<Field widget="text" id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a select field', () => {
    const component = renderer.create(<Field choices={[]} id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a boolean field', () => {
    const component = renderer.create(<Field type="boolean" id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a array field', () => {
    const component = renderer.create(<Field type="array" id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a fallback field', () => {
    const component = renderer.create(<Field id="test" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
