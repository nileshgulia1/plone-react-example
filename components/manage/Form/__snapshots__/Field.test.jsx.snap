// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders a array field 1`] = `
<div
  className="ArrayWidget"
/>
`;

exports[`Field renders a boolean field 1`] = `
<div
  className="CheckboxWidget"
/>
`;

exports[`Field renders a datetime field 1`] = `
<div
  className="DatetimeWidget"
/>
`;

exports[`Field renders a fallback field 1`] = `
<div
  className="TextWidget"
/>
`;

exports[`Field renders a password field 1`] = `
<div
  className="PasswordWidget"
/>
`;

exports[`Field renders a richtext field 1`] = `
<div
  className="WysiwygWidget"
/>
`;

exports[`Field renders a select field 1`] = `
<div
  className="SelectWidget"
/>
`;

exports[`Field renders a text field 1`] = `
<div
  className="TextWidget"
/>
`;

exports[`Field renders a textarea field 1`] = `
<div
  className="TextareaWidget"
/>
`;
